lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -156,11 +155,15 @@ def do_grant(kms, keyarn, role_arn, granttypes, mode='grant', dry_run=True, clea
             # do we want this grant type? Are we on its statement?
             # and does the role have this grant type?
 
-            # Ensure statement looks as expected
-            if not statement.get('Principal'):
-                statement['Principal'] = {'AWS': []}
-            if not isinstance(statement['Principal']['AWS'], list):
-                statement['Principal']['AWS'] = [statement['Principal']['AWS']]
+            # create Principal and 'AWS' so we can safely use them later.
+            if not isinstance(statement.get('Principal'), dict):
+                statement['Principal'] = dict()
+
+            if 'AWS' in statement['Principal'] and isinstance(statement['Principal']['AWS'], str):
```

Comment:
Instead of `str`, we need to allow for unicode:

```
    from ansible.module_utils.six import string_types
    if isinstance(x, string_types):
```

From the `CODING_GUIDELINES.md` doc